packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -82,7 +86,11 @@ class ErrorHandlingFileSystem extends ForwardingFileSystem {
       // Certain error codes indicate the file could not be found. It could have
       // been deleted by a different program while the tool was running.
       // if it still exists, the file likely exists on a read-only volume.
-      if (err.osError?.errorCode != kSystemCannotFindFile || _noExitOnFailure) {
+      // This check will falsely match "3/ESRCH: No such process" on Linux/macOS,
+      // but this should be fine since this code should never come up here.
```

Comment:
If this is a bit too smelly for your tastes, please read https://github.com/flutter/flutter/issues/150736#issuecomment-2187593847

Reply:
Why do you say "this code should never come up here"? Are you saying that we should never hit a "no such process" error on unix-like operating systems when calling `deleteIfExists`?

Reply:
(We talked about this in-person. I'm commenting here for posterity's sake). Yes, that's my hope. I'm assuming 3/ESRCH is never set by a file op. I'm assuming this code is used by Linux utilities that act on processes rather than files.